packages/async-flow/src/types.d.ts

```diff
@@ -78,15 +81,23 @@ export type GuestInterface<T> = {
  *
  * Specifically, Promise return values are converted to Vows.
  */
-export type HostOf<F extends CallableFunction> = F extends (
+export type HostFn<F extends CallableFunction> = F extends (
   ...args: infer A
 ) => infer R
   ? R extends Promise<infer T>
-    ? (...args: A) => Vow<T extends Passable ? T : HostInterface<T>>
-    : (...args: A) => HostInterface<R>
-  : F;
-
-export type HostArgs<GA extends any[]> = { [K in keyof GA]: HostOf<GA[K]> };
+    ? (...args: HostArgs<A>) => Vow<T extends Passable ? T : HostInterface<T>>
```

Comment:
Why not `Vow<T extends Passable ? HostOf<T> : never>` ?

Reply:
promises can return void, which is not Passable. I just pushed some alternatives

Reply:
In case it helps, `undefined` is Passable. Usually when I see `void` I think `undefined`, though I understand that `undefined` and `void` are distinct concepts in TS.

Reply:
ah yes https://github.com/endojs/endo/pull/2406